ansible/roles/test/files/ptftests/device_connection.py

```diff
@@ -32,9 +32,10 @@ def __init__(self, hostname, username, password=None, alt_password=None):
         '''
         self.hostname = hostname
         self.username = username
-        self.password = password
-        self.alt_password = alt_password
-        self.alt_password_index = 0
+        self.passwords = [password]
+        if alt_password:
+            self.passwords += alt_password
```

Comment:
```
$ python
Python 3.8.10 (default, Jan 17 2025, 14:40:23) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> 
>>> s = ["pass"]
>>> s += "pass2"
>>> s
['pass', 'p', 'a', 's', 's', '2']
>>> 

```

I don't think this is what you need. Please check?

Reply:
`alt_password` is expected to be a list of strings.

The new behavior is the same as the old behavior; if someone passes in just a string, then each individual character of that string will get used as the password. I can add a check/handling here to accept only lists, or if a string is passed in, either error out or wrap it into a list.